### ISSUE TITLE
Long periods of errors from service bus on single topic subscription

### ISSUE THREAD

**Issue by dandenton** [Azure/azure-sdk-for-net#14510]
While our application's service bus namespace had 0 incoming and outgoing messages, the receiver had the following exceptions for only a single topic/subscription:

```
Microsoft.Azure.ServiceBus.ServiceBusCommunicationException: The connection was inactive for more than the allowed 120000 milliseconds and is closed by container '7A2E0A0515A9427F82D1ACF7F25A024D_G28'.
---> Microsoft.Azure.ServiceBus.ServiceBusException: The connection was inactive for more than the allowed 120000 milliseconds and is closed by container '7A2E0A0515A9427F82D1ACF7F25A024D_G28'.
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.OnReceiveAsync(Int32 maxMessageCount, TimeSpan serverWaitTime)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.OnReceiveAsync(Int32 maxMessageCount, TimeSpan serverWaitTime)
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.<>c__DisplayClass64_0.<<ReceiveAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.ServiceBus.RetryPolicy.RunOperation(Func`1 operation, TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.RetryPolicy.RunOperation(Func`1 operation, TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.ReceiveAsync(Int32 maxMessageCount, TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.ReceiveAsync(TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.MessageReceivePump.<MessagePumpTaskAsync>b__11_0()
```

Following that we got these errors every 1-4 minutes for the next 45 minutes.

```
Microsoft.Azure.ServiceBus.ServiceBusException: The service was unable to process the request; please retry the operation. For more information on exception types and proper exception handling, please refer to http://go.microsoft.com/fwlink/?LinkId=761101 TrackingId:6d1857eb47f1430f9c359e7db7354d01_G28, SystemTracker:gateway7, Timestamp:2020-08-15T15:15:22
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.OnReceiveAsync(Int32 maxMessageCount, TimeSpan serverWaitTime)
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.<>c__DisplayClass64_0.<<ReceiveAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.ServiceBus.RetryPolicy.RunOperation(Func`1 operation, TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.RetryPolicy.RunOperation(Func`1 operation, TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.ReceiveAsync(Int32 maxMessageCount, TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.ReceiveAsync(TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.MessageReceivePump.<MessagePumpTaskAsync>b__11_0()
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.ReceiveAsync(TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.MessageReceivePump.<MessagePumpTaskAsync>b__11_0()
```
Our application is configured to retry using the built in exponential backoff using:
```
new RetryExponential(
                minimumBackoff: TimeSpan.FromSeconds(1),
                maximumBackoff: TimeSpan.FromSeconds(30),
                maximumRetryCount: 3);
```

All of the retries had the same TrackingID during this time.  Another time we got these exceptions we restarted the application and that resolved the issue.  We've opened a support ticket with Azure support and they only found transient errors on their side.  Because the errors went away after a restart they are suggesting it could be an SDK issue.  No other subscriptions were affected during this time.  So far this has only happened in our production environment (Standard service tier) and while nothing was using those topics.  We're not sure if these errors indicate that nothing would work over the topic or if they're just reporting transient errors that really have no affect on functionality.

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team best able to assist.

**Comment by axisc** [comment]
@dandenton It is normal to have the service disconnect from the idle connections. These are just transient errors because you have nothing to send/receive and process. 

Please remember that each QueueClient or SubscriptionClient is over an AMQP link which is a stateful bi-directional protocol layer connection that consumes compute. Closing idle links to preserve compute enables optimization of the resources.

hope this helps.

**Comment by axisc** [comment]
had a sync with the team and realized that this needs further investigation. Reopening this to track.

**Comment by dandenton** [comment]
Thanks @axisc.  I realize transient errors are expected but it is strange that 
1. The errors go on for long periods of time
2. They're resolved by restarting the application to restart the connection (isn't that what a retry of a transient error does if it's on the connection receiver)
3. They only happen on a single subscription and not any of the other 30-40

We're concerned about hiding transient errors because we don't want to miss issues with scalable resources if we're hitting rate limiting (e.g. HTTP 429) but we also don't want a lot of noise on our error logging channels that hides actual issues.

**Comment by yvgopal** [comment]
Could you also share the namespace and the entity that you were accessing? I need them to check logs on the service.

**Comment by dandenton** [comment]
@yvgopal It was namespace `safari-np-eastus`.  It happened twice on two different entities:

- 2020-08-15 between 15:10 and 15:32 UTC - `lpms.portalrecipientconfirmedreceipt/Subscriptions/Lpms` `TrackingId:6d1857eb47f1430f9c359e7db7354d01_G28, SystemTracker:gateway7, Timestamp:2020-08-15T15:15:22`

- 2020-08-09 between 20:31 and 21:14 UTC - `lpms.deleteallreaddbeventstreamsdata/Subscriptions/Lpms` `TrackingId:edd8b5d1c52549d0a65758e33f9cf636_G31, SystemTracker:gateway7, Timestamp:2020-08-09T20:35:26`

**Comment by yvgopal** [comment]
The correct namespace is 'safari-prd-eastus'. Good that both namespaces are hosted in the same cluster.

I have checked the service side logs and can confirm that it is a service side issue. The client SDK had no issues.
Between 11:30 UTC and 15;45 UTC on 8/15/2020, the node hosting your topic had experienced some issue that caused CPU usage to spike to 100%. CPU usage got really worse between 15:00 UTC and 15:45 UTC. Because of this high CPU, it refused all incoming connections. Around the same time your application was trying to connect to the topic/subscription and every connection attempt failed.

In standard tier, all entities of your namespace are not hosted on the same node. As only one of the nodes had this issue, requests to other entities likely not experienced the same issue.

We recently fixed an issue that was causing high CPU usage on nodes. Your issue might have been caused by it.
